client.go

```diff
@@ -102,7 +104,7 @@ func NewClient(config *Config) (client *Client, err error) {
 
 	tr := http.DefaultTransport.(*http.Transport)
 	tr.TLSClientConfig = &tls.Config{InsecureSkipVerify: config.SkipSslValidation}
-	ctx = context.WithValue(ctx, oauth2.HTTPClient, &http.Client{Transport: tr})
+	ctx = context.WithValue(ctx, oauth2.HTTPClient, &http.Client{Transport: tr, Timeout: config.Timeout})
```

Comment:
@mweibel,

I agree. I was hesitant to do that change because I didn't fully understand the reasoning behind the fact that `config.Client` is somewhat ignored in the coding. 

So, shall I update the pull request accordingly?

Best

Michael